proptest/src/arbitrary/_std/path.rs

```diff
@@ -9,15 +9,22 @@
 
 //! Arbitrary implementations for `std::path`.
 
-use std::path::*;
+use std::{ffi::OsString, path::*};
+
+use crate::{arbitrary::StrategyFor, prelude::Strategy, strategy::MapInto};
 
 // TODO: Figure out PathBuf and then Box/Rc/Box<Path>.
 
 arbitrary!(StripPrefixError; Path::new("").strip_prefix("a").unwrap_err());
 
+arbitrary!(PathBuf, MapInto<StrategyFor<OsString>, Self>;
+    OsString::arbitrary().prop_map_into()
+);
```

Comment:
agreed with @sunshowers comment that we should be a bit more robust in the impl here. I agree that a good starting place is to ensure we get a spread of path segments and we get coverage of absolute and relative paths. I'd be fine with merging a strategy that hits those two marks. I think more robust edge case generation could be done separately.

let me know if you'd be willing to tackle those in a follow up

Reply:
Unfortunately I won't have bandwidth to take this on probably for quite a while, but I hope this can be a very minimal bit of inspiration for others to build upon though.

Also I might point out that even though this is not an ideal implementation in terms of covering a wide range of possibilities, it *does* produce an arbitrary PathBuf, so maybe in the spirit of "something is better than nothing", this could be merged with the intention to improve upon it. Maybe even with a deprecation to make it clear that this is not a good implementation, but is currently the best we have? That would get me unstuck on my own project, where I really just need a completely arbitrary PathBuf, and have some types that contain PathBufs.

Reply:
Could either of you copy over the impl in camino? Otherwise I can try getting a PR up with that impl when I have time.

Reply:
Ended up doing this myself, see https://github.com/proptest-rs/proptest/pull/368.

Reply:
Thanks! I'm on vacation until next week so may not get a chance to look at it but will try to look sometime next week if I don't find any time before then